code/API_definitions/connected-network-type-subscriptions.yaml

```diff
@@ -0,0 +1,1202 @@
+openapi: 3.0.3
+info:
+  title: Connected Network Type Subscriptions
+  description: |
+    This API provides the customer with the ability to subscribe to notifications on changes to the connected network type of a device.
+
+    # Introduction
+
+    ## Connected Network Type Subscription
+    API consumers can subscribe to receive notifications when the connected network type of a device changes. These notifications are delivered through CloudEvents.
+
+    # Relevant terms and definitions
+
+    * **Device**: A device refers to any physical entity that can connect to a network and participate in network communication.
+      At least one identifier for the device (user equipment) out of four options: IPv4 address, IPv6 address, Phone number, or Network Access Identifier assigned by the mobile network operator for the device.
+
+    * **Connected Network Type**: Network Type is intended to provide insight into connected network's capabilities from standards perspective, and to reflect the mobile technology that would be displayed by the device to the end user where applicable.  Actual network capabilities may differ based on implementation and MUST be checked with the connected network provider.
+       - `2G`: if device is connected to the 2G network technology (alternative indicators such as "G" or "E" may be displayed on the device)
+       - `3G`: if device is connected to the 3G network technology (alternative indicators such as "H" or "H+" may be displayed on the device)
+       - `4G`: if device is connected to the 4G network technology (alternative indicators such as "LTE" or "LTE+" may be displayed on the device)
+       - `5G`: if device is connected to the 5G network technology
+       - `UNKNOWN`: if connection [technology] can not be determined
+    # API Functionality
+
+    The API exposes following capabilities:
+
+    ## Connected Network Type Subscription Management
+
+    These endpoints allow you to manage subscriptions for receiving notifications on changes to the connected network type of a device.
+    The CAMARA subscription model is detailed in the CAMARA API design guideline document and follows CloudEvents specification.
+
+    It is mandatory in the subscription to provide the event `type` for which the client would like to subscribe.
+
+    Following event ``type`` are managed for this API:
+      - ``org.camaraproject.connected-network-type-subscriptions.v0.network-type-changed``: This event is triggered when the connected network type of the device changes.
+
+    Note: Additionally, the event `org.camaraproject.connected-network-type-subscriptions.v0.subscription-ends` is sent when the subscription ends. This notification does not require a dedicated subscription.

```

Comment:
A detailed description of the cases when this subscription-ends event if fired is missing:

 It is used in following cases:
      - the subscription expire time (optionally set by the requester) has been reached
      - the maximum number of subscription events (optionally set by the requester) has been reached
      - the subscription was deleted by the requester
      - the Access Token `sinkCredential` (optionally set by the requester) expiration time has been reached
      - the API server has to stop sending notification prematurely